### ISSUE TITLE
Provide `code` strategy for verification flows

### ISSUE THREAD

**Issue by jonas-jonas** [ory/kratos#2760]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [X] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

After #2645 is merged, there is a new `code` strategy available for recovery, but not for verification flows.

### Describe your ideal solution

The new strategy needs to be available for verification flows as well.

- New config key `selfservice.flows.verification.use` with options `"link" | "code"`
- New email templates for `verification_valid` and `verification_invalid`
- New strategy to fulfill verification requests

### Workarounds or alternatives

none.

### Version

v0.10.1

### Additional Context

#2645

**Comment by jonas-jonas** [comment]
Superseded by https://github.com/ory/kratos/issues/2824, whoops.

**Comment by jonas-jonas** [comment]
Superseded by https://github.com/ory/kratos/issues/2824, whoops.
